rotkehlchen/tests/exchanges/test_htx.py

```diff
@@ -71,3 +99,128 @@ def test_assets_are_known(htx_exchange: Htx, globaldb):
+    mock_fn = htx_account_mock(calls={
+        '/v1/order/matchresults': [
+            json.loads('[{"symbol": "unibotusdt", "fee-currency": "unibot", "source": "spot-web", "filled-amount": "0.604608772660971035", "filled-fees": "0.001209217545321942", "filled-points": "0.0", "role": "taker", "updated-at": null, "price": "38.8719", "created-at": 1710170117673, "fee-deduct-currency": "", "fee-deduct-state": "done", "order-id": 1020938505066467, "match-id": 612882223, "trade-id": 30318429, "id": 1021138748019697, "type": "buy-market"}, {"symbol": "unibotusdt", "fee-currency": "unibot", "source": "spot-web", "filled-amount": "3.0035", "filled-fees": "0.006007", "filled-points": "0.0", "role": "taker", "updated-at": null, "price": "38.8719", "created-at": 1710170117673, "fee-deduct-currency": "", "fee-deduct-state": "done", "order-id": 1020938505066467, "match-id": 612882223, "trade-id": 30318428, "id": 1021138748019696, "type": "buy-market"}, {"symbol": "unibotusdt", "fee-currency": "unibot", "source": "spot-web", "filled-amount": "1.537", "filled-fees": "0.003074", "filled-points": "0.0", "role": "taker", "updated-at": null, "price": "38.8718", "created-at": 1710170117672, "fee-deduct-currency": "", "fee-deduct-state": "done", "order-id": 1020938505066467, "match-id": 612882223, "trade-id": 30318427, "id": 1021138748019698, "type": "buy-market"}, {"symbol": "daiusdt", "fee-currency": "usdt", "source": "spot-web", "filled-amount": "1038.18", "filled-fees": "2.07532182", "filled-points": "0.0", "role": "taker", "updated-at": null, "price": "0.9995", "created-at": 1710154700452, "fee-deduct-currency": "", "fee-deduct-state": "done", "order-id": 1020936358404315, "match-id": 100304520643, "trade-id": 7343200, "id": 1021136634087546, "type": "sell-market"}, {"symbol": "daiusdt", "fee-currency": "usdt", "source": "spot-web", "filled-amount": "26.3", "filled-fees": "0.05257896", "filled-points": "0.0", "role": "taker", "updated-at": null, "price": "0.9996", "created-at": 1710154700451, "fee-deduct-currency": "", "fee-deduct-state": "done", "order-id": 1020936358404315, "match-id": 100304520643, "trade-id": 7343199, "id": 1021136634087547, "type": "sell-market"}]'),  # noqa: E501
+        ],
+    })
+    with patch.object(htx_exchange, '_query', side_effect=mock_fn):
+        trades, _ = htx_exchange.query_online_trade_history(
+            start_ts=Timestamp(1710154700),
+            end_ts=Timestamp(1710170118),
+        )
+    expected_trades = [
+        Trade(
+            timestamp=Timestamp(1710170117),
+            location=Location.HTX,
+            base_asset=Asset('eip155:1/erc20:0xf819d9Cb1c2A819Fd991781A822dE3ca8607c3C9'),
+            quote_asset=A_USDT,
+            trade_type=TradeType.BUY,
+            amount=AssetAmount(FVal('0.604608772660971035')),
+            rate=Price(FVal('38.8719')),
+            fee=Fee(FVal('0.001209217545321942')),
+            fee_currency=Asset('eip155:1/erc20:0xf819d9Cb1c2A819Fd991781A822dE3ca8607c3C9'),
```

Comment:
how do you know if the fee is in the sell or the buy side? How you know is the unibot asset here?

Reply:
It's mentioned in the docs here, https://www.htx.com/en-us/opend/newApiPages/?id=7ec4de21-7773-11ed-9966-0242ac110003

```
Currency of transaction fee or transaction fee rebate (transaction fee of buy order is based on base currency, transaction fee of sell order is based on quote currency; transaction fee rebate of buy order is based on quote currency, transaction fee rebate of sell order is based on base currency)
```


Will add a comment in the function.